app/assets/javascripts/discourse/app/lib/to-markdown.js

```diff
@@ -48,6 +51,30 @@ export function clearBlockDecorateCallbacks() {
   blockDecorateCallbacks = [];
 }
 
+/**
+ * Allows to add support for custom text node transformations
+ * based on the next/previous elements.
+ *
+ * ```
+ * addTextDecorateCallback(function (text, nextElement, previousElement) {
+ *   if (
+ *     startRangeOpts &&
+ *     nextElement?.attributes.class?.includes("discourse-local-date") &&
+ *     text === "→"
+ *   ) {
+ *     return "";
+ *   }
+ * });
+ * ```
+ */
+export function addTextDecorateCallback(callback) {
+  textDecorateCallbacks.push(callback);
+}
+
+export function clearTextDecorateCallbacks() {
+  textDecorateCallbacks = [];
+}
+
```

Comment:
Does this belong in the plugin js api, at least so we call out it is an available extensibility point?

Reply:
Yes that would be better, I can move all three of these there and possibly name them a little differently, I think they are only used by one of our plugins ATM so I can change it there easily